app/src/main/java/com/asfoundation/wallet/promotions/ui/PromotionsFragment.kt

```diff
@@ -246,6 +249,7 @@ class PromotionsFragment : BasePageViewFragment(),
   }
 
   private fun getGamificationHeaderBinding(gamificationHeaderItem: GamificationItem): GamificationHeaderBindingAdapter {
+    hideAllGamificationLayouts()
```

Comment:
Not a good idea to have side effects in a `get...` function. Mb move it into `showPromotionsHeader` function.

Reply:
agreed. changed